packages/plugins/i18n/server/register.js

```diff
@@ -9,11 +9,51 @@ const enableContentType = require('./migrations/content-type/enable');
 const disableContentType = require('./migrations/content-type/disable');
 
 module.exports = ({ strapi }) => {
+  decorateRelations();
   extendLocalizedContentTypes(strapi);
   addContentManagerLocaleMiddleware(strapi);
   addContentTypeSyncHooks(strapi);
 };
 
+/**
+ * Wraps the /relations controller to handle locale parameter
+ */
+const decorateRelations = () => {
+  const { wrapParams } = getService('entity-service-decorator');
+
+  strapi.container.get('controllers').extend('plugin::content-manager.relations', (controller) => {
+    const oldFindAvailable = controller.findAvailable;
+    return Object.assign(controller, {
+      async findAvailable(ctx, next) {
+        const { model, targetField } = ctx.params;
+        const { component } = ctx.request.query;
+
+        const sourceModelUid = component || model;
+
+        const sourceModel = strapi.getModel(sourceModelUid);
+        if (!sourceModel) {
+          return ctx.badRequest("The model doesn't exist");
+        }
+
+        const attribute = sourceModel.attributes[targetField];
+        if (!attribute || attribute.type !== 'relation') {
+          return ctx.badRequest("This relational field doesn't exist");
+        }
+
+        const targetedModel = strapi.getModel(attribute.target);
+
+        const { isLocalizedContentType } = getService('content-types');
+
+        if (isLocalizedContentType(targetedModel)) {
+          ctx.request.query = await wrapParams(ctx.request.query);
```

Comment:
I thought this param was already coming from the frontend request 🤔

Reply:
It transforms `?locale=en` into filters and put the default locale if none are set

Reply:
So it's because we are not using the entity service anymore that we need then 🤔 now that you can use the findPage I'm even wondering if you could use the entityService to avoid having to do this change at all 🤔  wdyt ?

Reply:
Sure let's try when you are available